core/src/main/java/com/devonfw/tools/solicitor/componentinfo/scancode/UncuratedScancodeComponentInfoProvider.java

```diff
@@ -0,0 +1,275 @@
+            }
+          }
+          // do not output the Classpath-exception-2.0 as separate License
+          spdxIdMap.remove("Classpath-exception-2.0");
+        }
+      }
+      for (JsonNode li : file.get("licenses")) {
+        String licenseid = li.get("key").asText();
+        String licenseName = li.get("spdx_license_key").asText();
+        String effectiveLicenseName = spdxIdMap.get(licenseName);
+        if (effectiveLicenseName == null) {
+          // not contained in map --> this must be the Classpath-exception-2.0
+          continue;
+        } else {
+          licenseName = effectiveLicenseName;
+          if (licenseName.endsWith("WITH Classpath-exception-2.0")) {
+            licenseid = licenseid + "WITH Classpath-exception-2.0";
+          }
+        }
+        String licenseDefaultUrl = li.get("scancode_text_url").asText();
+        licenseDefaultUrl = normalizeLicenseUrl(packageUrl, licenseDefaultUrl);
+        double score = li.get("score").asDouble();
+        String licenseFilePath = file.get("path").asText();
+        int startLine = li.get("start_line").asInt();
+        int endLine = li.get("end_line").asInt();
+        if (!takeCompleteFile) {
+          licenseFilePath += "#L" + startLine;
+          if (endLine != startLine) {
+            licenseFilePath += "-L" + endLine;
+          }
+        }
+
+        licenseFilePath = normalizeLicenseUrl(packageUrl, licenseFilePath);
+        String givenLicenseText = null;
+        if (licenseFilePath != null && licenseFilePath.startsWith(ComponentContentProvider.PATH_PREFIX)) {
+          givenLicenseText = this.fileScancodeRawComponentInfoProvider.retrieveContent(packageUrl, licenseFilePath);
+        }
+
+        componentScancodeInfos.addLicense(licenseid, licenseName, licenseDefaultUrl, score, licenseFilePath,
+            givenLicenseText, endLine - startLine);
+      }
+    }
+    // if (componentScancodeInfos.getLicenses().size() == 0) {
+    // componentScancodeInfos.addLicense("unknown", "unknown", "", 100.0, "", "", 0);
+    // }
+    if (componentScancodeInfos.getNoticeFilePath() != null
+        && componentScancodeInfos.getNoticeFilePath().startsWith(ComponentContentProvider.PATH_PREFIX)) {
+      componentScancodeInfos.setNoticeFileContent(this.fileScancodeRawComponentInfoProvider.retrieveContent(packageUrl,
+          componentScancodeInfos.getNoticeFilePath()));
+    }
+    return componentScancodeInfos;
+  }
+
+  /**
+   * Adjustment of license paths/urls so that they might retrieved
+   *
+   * @param packageUrl package url of the package
+   * @param licenseFilePath the original path or URL
+   * @return the adjusted path or url as a url
+   */
+  private String normalizeLicenseUrl(String packageUrl, String licenseFilePath) {
+
+    String adjustedLicenseFilePath;
+    if (licenseFilePath != null) {
+      if (licenseFilePath.startsWith("http")) {
+        // TODO
```

Comment:
what is the TODO here?

Reply:
I don't know anymore... this code is to a large degree heuristic/experimental. I deleted the TODO. The business code on how mapping is done needs some overhaul anyway.